### PR TITLE
SRW-10: Create `<ProfileHeader />`

### PR DESCRIPTION
**Pull request by erunks**
## What
Create a `<ProfileHeader />` component for  the profile page that'll be built out eventually. Also built out some sub components, that'll be reused later, and some placeholder components.

## Link to Issue
Resolves https://github.com/speedrun-website/speedrun.website/issues/10

### Steps for testing

- Checkout the PR and add it to a page, in order to see it in action

### Images

| • | Landscape | Portrait |
| --- | --- | --- |
| Mobile | Screen Shot 2021-08-16 at 09 01 39 Screen Shot 2021-08-16 at 09 01 42 | Screen Shot 2021-08-16 at 09 01 23 |
| Tablet | Screen Shot 2021-08-16 at 09 02 04 | Screen Shot 2021-08-16 at 09 02 02 |
| Desktop | Screen Shot 2021-08-16 at 09 02 18 | Screen Shot 2021-08-16 at 09 03 31 |

### CODE REVIEW

File: components/ProfileHeader/ProfileHeader.vue [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        </div>
+        <div class="profile-info--user-bio">
+          <h1 class="text-2xl font-semibold">{{ username }}</h1>
+          <h2 class="text-sm text-black text-opacity-50">{{ location }}</h2>
+          <p class="text-black text-opacity-50 pt-2 md:hidden">{{ bio }}</p>
+        </div>
+        <div class="profile-info--follow">
+          <CoreFollowButton />
+        </div>
+        <div class="profile-info--user-activity hidden md:grid">
```
Comment (legrandgrand) [inline components/ProfileHeader/ProfileHeader.vue:28]: this is repeated stuff compared to what is above. Could you make a small component for that part that would only be used in the profileHeader for better visibility and avoid repetition?

File: components/core/FollowButton/FollowButton.test.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import { fireEvent, stubbedRender } from '@/testUtils';
+import FollowButton from './FollowButton';
+
+describe('<FollowButton />', () => {
+  const mockOnClick = jest.fn();
```
Comment (legrandgrand) [inline components/core/FollowButton/FollowButton.test.js:5]: It took me a while to understand what mockOnClick referred to on line 44. Maybe rename mockOnClick to jestFn ? or maybe inline everything and don't use global variables? Not sure what the best way to do it.

**Review by legrandgrand — CHANGES_REQUESTED** [review]
Very minor stuff to review. Good stuff still, and glad to see some tests :D

File: components/ProfileHeader/ProfileHeader.vue (hunk 1 above)
Comment (erunks) [reply components/ProfileHeader/ProfileHeader.vue:28]: Yeah, I can break that out for sure.

File: components/core/FollowButton/FollowButton.test.js (hunk 2 above)
Comment (erunks) [reply components/core/FollowButton/FollowButton.test.js:5]: Would it be more helpful to leave some comments around some of the setup? This is a convention of mocking functions in tests, at least in React. Not sure what kind of conventions Vue uses for testing though.

File: components/ProfileHeader/ProfileHeader.vue (hunk 1 above)
Comment (erunks) [reply components/ProfileHeader/ProfileHeader.vue]: Addressed that with this: https://github.com/speedrun-website/speedrun.website/pull/34/commits/6a726581feaa8c3a4721564035778df65bae3b5a

File: components/core/FollowButton/FollowButton.test.js (hunk 2 above)
Comment (legrandgrand) [reply components/core/FollowButton/FollowButton.test.js:5]: If that's convention that's fine by me!

**Review by legrandgrand — APPROVED** [review]

**Pull request merged**